Demos: Replace CSS code samples with SCSS includes

EricDunsworth opened the pull request:
The equal heights demo's hardcoded CSS sample was outdated compared to its "real" SCSS source file.

Rather than updating the affected CSS sample, this replaces all demo CSS samples with future-proof SCSS ones. The SCSS samples are setup as includes (handlebars partials) that are sourced from the demos' "real" SCSS files. #9566's custom handlebars helpers are also used to make the SCSS samples more presentable.

EricDunsworth commented:
FWIW I initially wanted to stick with CSS samples and derive them from the dist folder's unminified CSS files... but ran into snags while testing. Build failures could potentially occur if the demos were compiled before the demo CSS files (like running ``grunt demos`` without a pre-existing dist folder).

duboisp commented:
This seem to be a content change only. Approved upon review and local testing.

Garneauma reviewed (changes requested):
@EricDunsworth Please fix merge conflicts.

EricDunsworth committed:
Demos: Replace CSS code samples with SCSS includes

The equal heights demo's hardcoded CSS sample was outdated compared to its "real" SCSS source file.

Rather than updating the affected CSS sample, this replaces all demo CSS samples with future-proof SCSS ones. The SCSS samples are setup as includes (handlebars partials) that are sourced from the demos' "real" SCSS files. #9566's custom handlebars helpers are also used to make the SCSS samples more presentable.

EricDunsworth commented:
@Garneauma Just rebased and updated modified dates 👍.

Garneauma reviewed (approved):
Tested locally and everything works as expected. Thank you @EricDunsworth for your contribution!
